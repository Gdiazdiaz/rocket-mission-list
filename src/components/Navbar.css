* {
  margin: 0;
  padding: 0;
  box-sizing: border-box;
}

body {
  background-image: url("../media/R.jpg");
}

/* .background-pic{
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  z-index: -2;
  background-size: contain;
} */

.inactive {
  color: #fff;
  text-decoration: none;
}

.active {
  text-decoration: underline;
  color: #fff;
}

.logo-container {
  display: flex;
  align-items: center;
  gap: 10px;
  padding-top: 10px;
  padding-left: 10px;
  padding-bottom: 10px;
  color: #fff;
}

.logo-navbar {
  width: 50px;
}

.navbar-main-container {
  display: flex;
  justify-content: space-between;
  align-items: center;
  background-color: #4c3b7180;
}

.nav-items {
  display: flex;
  list-style: none;
  gap: 20px;
  padding-right: 10px;
}
