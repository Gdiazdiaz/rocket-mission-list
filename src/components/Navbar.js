import React from 'react';
import { NavLink } from 'react-router-dom';
import './Navbar.css';
import pic from '../media/white-planet.png';

function Navbar() {
  return (
    <nav className="navbar-main-container">
      <div className="logo-container">
        <img className="logo-navbar" alt="Logo" src={pic} />
        <h1>Rocket Mission</h1>
      </div>
      <div>
        <ul className="nav-items">
          <li><NavLink to="/RocketsContainer" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>Rockets</NavLink></li>
          <li><NavLink to="/Missions" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>Missions</NavLink></li>
          <li><NavLink to="/" className={({ isActive }) => (isActive ? 'active' : 'inactive')}>My Profile</NavLink></li>
        </ul>
      </div>
    </nav>
  );
}

export default Navbar;
