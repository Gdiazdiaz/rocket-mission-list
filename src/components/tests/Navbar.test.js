import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Navbar from '../Navbar';
import '@testing-library/jest-dom';

describe('Checking page deployment', () => {
  test('Navbar component matches snapshot', () => {
    const tree = renderer.create(<Router><Navbar /></Router>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});