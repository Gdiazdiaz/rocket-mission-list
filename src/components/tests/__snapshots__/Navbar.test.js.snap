// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checking page deployment Navbar component matches snapshot 1`] = `
<nav
  className="navbar-main-container"
>
  <div
    className="logo-container"
  >
    <img
      alt="Logo"
      className="logo-navbar"
      src="white-planet.png"
    />
    <h1>
      Rocket Mission
    </h1>
  </div>
  <div>
    <ul
      className="nav-items"
    >
      <li>
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          Rockets
        </a>
      </li>
      <li>
        <a
          className="inactive"
          href="/Missions"
          onClick={[Function]}
        >
          Missions
        </a>
      </li>
      <li>
        <a
          className="inactive"
          href="/Profile"
          onClick={[Function]}
        >
          My Profile
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
